DEV-INTRO.md

```diff
@@ -1,59 +1,76 @@
-PEcAn Development
-=================
+# PEcAn Development
 
-Directory Structure
--------------------
+This is a minimal guide to getting started with PEcAn development.
 
-### pecan/
+## Git Repository and Workflow
 
-* modules/  Contains the modules that make up PEcAn
-* web/		The PEcAn web app to start a run and view outputs.
-* models/		Code to create the model specific configurations.
-* documentation/	Documentation about what PEcAn is and how to use it.
+The recommended workflow is gitflow which is described in the PEcAn documentation. For the repository we recommend using a [fork of the PEcAn repsitory](https://help.github.com/en/github/getting-started-with-github/fork-a-repo). In the scripts folder there is a script called [syngit.sh](scripts/syncgit.sh) that will help with synchronizing your fork with the official repository.
 
-### Modules (R packages)
+## Developing in Docker
 
-* General
-** all
-** utils
-** db
-* Analysis
-** modules/meta.analysis 
-** modules/uncertainty"
-** modules/data.land 
-** modules/data.atmosphere
-** modules/assim.batch 
-** modules/assim.sequential 
-** modules/priors
-* Model Interfaces
-** models/ed 
-** models/sipnet 
-** models/biocro
+To get started with development in docker we need to bring up the docker stack first. In the main pecan folder you will find the [docker-compose.yml](docker-compose.yml) file that can be used to bring up the pecan stack. There is also the [docker-compose.dev.yaml](docker-compose.dev.yaml) file that adds additional containers, and changes some services to make it easier for development.
 
+You can copy the `docker-compose.dev.yaml` to `docker-compose.override.yml`. Once that is done the `docker-compose` program will automatically use the `docker-compose.yml` and `docker-compose.override.yml` files, and you don't have to explicitly load them in the commands below.
 
-#### List of modules 
+### First time setup
 
-Installing PEcAn
-----------------
+You can copy the [`env.example`](docker/env.example) file as .env in your pecan folder. The variables we want to modify are:
 
-### Virtual Machine
+* `COMPOSE_PROJECT_NAME` set this to pecan, the prefix for all containers
+* `PECAN_VERSION` set this to develop, the docker image we start with
 
-* Fastest way to get started
-* see PEcAn demo ...
+Next we will create the folders that will hold all the data for the docker containers using: `mkdir -p volumes/{lib,pecan,portainer,postgres,rabbitmq,traefik}`. The `volumes` folder will be ignored by git.
```

Comment:
As someone who hasn't make the transition from command line to docker, these parts are confusing. At a high level, (1) it's not clear why all this needs to be done and (2) if we need all this stuff why isn't it there already in the repo (e.g. why would we need to modify values in an example file, shouldn't the example just be set to the right thing? Does .env impact non-Docker installs, and if not why not just have .env there by default?) At a lower level, when the md is is rendered, will it be obvious where the env.example is located? Also, the volumes gets added in the pecan folder? (just trying to minimize places someone could mess up).

Also, it's not 100% clear to me whether the next set of steps using docker-composer are really 
"one-time" or things that need to be done every time you want to start up your pecan docker stack

Reply:
Added some more text to add some notes about the folders etc. The links in the document right now assume it is in the root of the pecan git repo, and point to the files in the git repo.